src/Honeybee_IntersectMasses.py

```diff
@@ -50,173 +50,68 @@
-                    jlCenter = jlProps.Centroid
-                    jlArea = jlProps.Area
-                    faceMatchInit = False
-                    for face in building.Faces:
-                        faceProps = rc.Geometry.AreaMassProperties.Compute(face)
-                        faceCent = faceProps.Centroid
-                        faceArea = faceProps.Area
-                        if faceCent.DistanceTo(jlCenter) < sc.doc.ModelAbsoluteTolerance and jlArea-faceArea < sc.doc.ModelAbsoluteTolerance and jlArea-faceArea > -sc.doc.ModelAbsoluteTolerance:
-                            faceMatchInit = True
-                else:
-                    faceMatchInit = False
-                
-                if faceMatchInit == False:
-                    #Try splitting the faces of the brep with the intersection curve.
-                    newBldgBreps = []
-                    for srf in building.Faces:
-                        newBrep = srf.Split([joinedLine], sc.doc.ModelAbsoluteTolerance)
-                        if newBrep.Faces.Count > building.Faces.Count:
-                            if isPlanar == True:
-                                faceMatch = False
-                                for face in newBrep.Faces:
-                                    faceProp2 = rc.Geometry.AreaMassProperties.Compute(face)
-                                    faceCent2 = faceProp2.Centroid
-                                    faceArea2 = faceProp2.Area
-                                    if faceCent2.DistanceTo(jlCenter) < sc.doc.ModelAbsoluteTolerance and jlArea-faceArea2 < sc.doc.ModelAbsoluteTolerance and jlArea-faceArea2 > -sc.doc.ModelAbsoluteTolerance:
-                                        faceMatch = True
-                                if faceMatch == True:
-                                    newBldgBreps.append(srf.Split([joinedLine], sc.doc.ModelAbsoluteTolerance))
-                            else:
-                                faceMatch = False
-                                for face in newBrep.Faces:
-                                    srfEq = isSrfGeoEquivalent(joinedLine, face)
-                                    if srfEq == True and face.IsPlanar() == True:
-                                        faceMatch = True
-                                    elif srfEq == False and face.IsPlanar() == False:
-                                        faceMatch = True
-                                if faceMatch == True:
-                                    newBldgBreps.append(srf.Split([joinedLine], sc.doc.ModelAbsoluteTolerance))
-                    
-                    try:
-                        newBuilding = rc.Geometry.Brep.JoinBreps(newBldgBreps, sc.doc.ModelAbsoluteTolerance)[0]
-                        building = newBuilding
-                    except:
-                        pass
-    
-    return building
+    changed = False
+
+    joinedLines = []
+    i = 0
+    for i in range(500):
```

Comment:
What is it that makes 500 the right number here?  Shouldn't this be a while loop if you want it to exit after a certain condition is met?

Reply:
This is to prevent a dead `while` loop. This loop will break if `building` and `otherBldg` have no intersection. I prefer to use `for` with a large number instead of `while` to prevent the program to crush.

Reply:
@Larry7707 ,
Ok but doesn't this mean that the component won't be able to find an intersection when there are more that 500 surfaces to intersect?  Why not set it to iterate over all the surfaces of the otherBldg instead of an arbitrary number?

Reply:
Hi @chriswmackey ,
No. the `for` loop here is just to rerun the intersect procedure to make sure that there is no more intersection between `building` and `otherBldg`. The `for` loop here is just for double-check purpose, like an insurance. The `for` loop below actually does all the intersection work. 
```
for face1 in building.Faces:
            if face1.IsSurface:
                # if it is a untrimmed surface just find intersection lines
                intersectLines = rc.Geometry.Intersect.Intersection.BrepSurface(otherBldg, face1.DuplicateSurface(), tol)[1]
            else:
                # if it is a trimmed surface
                edgesIdx = face1.AdjacentEdges()
                edges = []
                for ix in edgesIdx:
                    edges.append(building.Edges.Item[ix])
                crv = rc.Geometry.Curve.JoinCurves(edges, tol)
                # potential bug - a surface has made of multiple crvs, but not likely
                intersectLines = rc.Geometry.Intersect.Intersection.BrepBrep(rc.Geometry.Brep.CreatePlanarBreps(crv[0])[0], otherBldg, tol)[1]
            temp = rc.Geometry.Curve.JoinCurves(intersectLines, tol)
            joinedLines = [crv for crv in temp if rc.Geometry.Brep.CreatePlanarBreps(crv)]
            if len(joinedLines) > 0:
                newBrep = face1.Split(joinedLines, tol) # return None on Failure
                if not newBrep: continue
                if newBrep.Faces.Count > building.Faces.Count:
                    changed = True
                    building = newBrep
                    break
```
For your second question, the function used every `face` from `building` to intersect with `otherBldg` as a brep so that it will prevent non-planar problem while minimizing the operation to reconstruct trimmed faces. Rhino BrepFace is a little bit tricky to handle, especially trimmed faces. If we use faces from `building` to intersect every face from `otherBldg`, it will be too slow.

Reply:
@larry7707 ,
I understand what you have written now but I still think that it should be done with a while loop rather than a for loop.  The function you have here will fail if there are more than 500 intersections between one building and another and, while this situation is very rare, there's no need to introduce this arbitrary limitation.

I think you should have a variable that is set to False at the start and, at the end of each iteration of the while loop, you check whether tempBldg.Faces.Count == building.Facea.Count. if it does, then this varibale gets set to True and the while loop ends. So, instead of a for loop over 500, you have "while variable is False: keep looking for intersections"

Once you make this cahnge, I will merge in what you have here.

Reply:
@chriswmackey ,
Yea I think you are right. I have changed the `for` loop and added a variable `done` as the flag. 
Thanks for your comment and also let me know if there are any other questions when merging.
Happy New Year!